src/zh/baimangu/src/eu/kanade/tachiyomi/extension/zh/baimangu/Baimangu.kt

```diff
@@ -0,0 +1,339 @@
+        val manga = SManga.create().apply {
+            title = element.select("dd.fed-deta-content a:first-child").first().text()
+            thumbnail_url = picElement.attr("data-original")
+        }
+
+        manga.setUrlWithoutDomain(picElement.attr("href"))
+
+        return manga
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val picElement = document.select("a.fed-list-pics").first()
+        val picUrl = picElement.attr("data-original")
+        val detailElements = document.select("dd.fed-deta-content ul.fed-part-rows")
+        return SManga.create().apply {
+            title = document.select("h1.fed-part-eone").first().text().trim()
+
+            if (!invalidCoverImageDomains.any { picUrl.contains(it) }) {
+                thumbnail_url = picUrl
+            }
+
+            // They don't seem to show the status, unless we query from the category, but that may be inaccurate
+            status = SManga.UNKNOWN
+
+            author = detailElements.select("li:nth-child(1) a").firstOrNull()?.text()?.trim()
+
+            genre = detailElements.select("li.fed-show-md-block:nth-last-child(2) a:not(:empty)")?.joinToString { it.text().trim() }
+
+            // It has both "简介：" and "简介" in the description
+            description = detailElements.select("li.fed-show-md-block:nth-last-child(1)")
+                .firstOrNull()?.text()?.replace("简介：", "", ignoreCase = true)
+                ?.replace("简介", "", ignoreCase = true)?.trim()
+        }
+    }
+
+    override fun chapterListSelector(): String = "div.fed-play-item ul.fed-part-rows:last-child a"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val chapter = SChapter.create().apply {
+            name = element.text().trim()
+        }
+        chapter.setUrlWithoutDomain(element.attr("href"))
+        return chapter
+    }
+
+    override fun imageUrlParse(document: Document) = ""
+
+    override fun pageListParse(document: Document): List<Page> {
+        val oScriptUrl = extractOScriptUrl(document)
+
+        val oScriptResp = client.newCall(GET(oScriptUrl)).execute()
+
+        if (!oScriptResp.isSuccessful) {
+            throw Error("Failed to request OScript URL")
+        }
+
+        val content = oScriptResp.body!!.string()
+        return extractPagesFromOScript(content)
+    }
+
+    private fun extractOScriptUrl(document: Document): String {
+        val scriptElements = document.getElementsByTag("script")
+        for (element in scriptElements) {
+            val data = element.data()
+            if (data.contains("oScript.src")) {
+                val pattern = Pattern.compile("src(\\s*)=(\\s*)\"(.+)\";")
+                val matcher = pattern.matcher(data)
+                if (matcher.find()) {
+                    return matcher.group(3) ?: throw Exception("Unable to extract OScript")
+                }
+            }
+        }
+
+        throw Error("Unable to match for OScript")
```

Comment:
No need to iterate through all scripts, just directly find the one with Jsoup's `:containsData` selector.

```suggestion
        val theScript = document.selectFirst("script:containsData(oScript.src)")?.data()
            ?: throw Exception("Unable to find the OScript")
            
        val pattern = Pattern.compile("src(\\s*)=(\\s*)\"(.+)\";")
        val matcher = pattern.matcher(data)
        
        if (matcher.find()) {
            return matcher.group(3) ?: throw Exception("Unable to extract OScript")
        }

        throw Error("Unable to find a match for OScript")
```

Please check if it's still working after changing it.

Reply:
Nice, didn't know we can do `script:containsData`